src/FsAutoComplete/FsAutoComplete.Lsp.fs

```diff
@@ -1774,8 +1774,12 @@ type FsharpLspServer(commands: Commands, lspClient: FSharpLspClient) =
           match res with
           | None -> LspResult.internalError "No highlights found"
           | Some res ->
-            { Content = CommandResponse.highlighting FsAutoComplete.JsonSerializer.writeJson res }
-            |> success
+            let mapped =
```

Comment:
it's way easier to do tests if we have the endpoint return a typed response (counting on the LSP layer to do the serialization. Is this alright?

Reply:
I'm not sure about this - it will probably break deserialization on the client-side. Highlighting is not really part of LSP yet (?) so this is our custom protocol, and I think it'd be better to keep it consistent with our other custom endpoints

Reply:
That said, probably in the future we should improve this for all of our custom endpoints, the way it is right now has some legacy reasons, but not really any valid right now.

Reply:
Yeah, if we took this change I'd have to patch up ionide as well, and of course I'd do that. I do think it's worth doing, even if it is piecemeal.

Reply:
Although if I am looking at changing the public contract here, I should remove the FCS range from the contract in favor of LSP positions I bet.

Reply:
I reverted this API change for now. On the tests' side of things I just do the deserialization there into the typed structure, works ok.